onnxruntime/core/providers/op_kernel_type_control.h

```diff
@@ -84,15 +85,15 @@ struct EnabledTypes {
   template <typename T>
   using GetTypesMember = typename T::types;
 
-  // checks whether T has data member 'types'
+  // checks whether T has a type alias member called 'types'
   template <typename T>
   using HasTypesMember = boost::mp11::mp_valid<GetTypesMember, T>;
 
   static_assert(HasTypesMember<SupportedTypesHolder>::value,
-                "SupportedTypesHolder must have a 'types' data member.");
+                "SupportedTypesHolder must have a type alias called 'types'.");
 
   // the allowed type lists to consider
-  // for each element of AllowedTypesHolders, get and include a 'types' data member if present
+  // for each element of AllowedTypesHolders, get and include the content of the 'types' type alias member if present
```

Comment:
>the content of [](start = 62, length = 14)

nit: does "the content of" mean the contained types? it's actually getting the type list itself. e.g., given:
`types = TypeList<int, float>`
we extract `TypeList<int, float>` vs. `int, float`

long way of suggesting: "get and include the 'types' type alias member"